### DISCUSSION TITLE
Shooting methods

### DISCUSSION THREAD

**Discussion by ke7kto** [idaholab/moose#17477] [Q&A Modules: General] [answered]
It seems that this might be a little out of scope, but is there anything like a [shooting method](https://en.wikipedia.org/wiki/Shooting_method) within moose, where one could vary, e.g. a material property, with a boundary condition acting as a constraint? I would imagine this could be highly useful, but also a very heavy lift to implement.

**Comment by GiudGiud** [comment, marked as answer]
Hello

I have a PR #17420 up for performing small optimization problems like this using the Control system, and more precisely a PID controller in a pseudo-transient calculation.

Merging is likely to be pretty delayed as I want to add more capability to this but you could try it out.

Guillaume

**Reply by ke7kto** [reply]
I'll take a look. Thanks!
